hypervisor/src/hypervisor/partition.rs

```diff
@@ -275,6 +277,24 @@ impl Run {
                 //umount("old").unwrap();
                 chdir("/").unwrap();
 
+                // After we've performed the pseudo chroot, we can create the
+                // Unix domain socket
+                let syscall_socket = socket::socket(
+                    AddressFamily::Unix,
+                    SockType::Datagram,
+                    SockFlag::SOCK_CLOEXEC,
+                    None,
+                )
+                .unwrap();
+
+                bind(syscall_socket, &UnixAddr::new("/syscall").unwrap()).unwrap();
+
+                // Now that the socket is up and running, we will create a satellite thread
+                // listening on it.
+                std::thread::spawn(move || {
+                    syscall::handle(syscall_socket, None).unwrap();
+                });
```

Comment:
I would prefer if we could avoid threads and instead use mio to listen to all the uds of the currently running threads + timer for the next schedule event (e.g. end/begin of a partition window). However, if this works, I'm fine just putting a `// TODO use mio instead of threads`.

Reply:
I am somewhat indifferent about threads here as well. My original idea was to spawn a child process as something like a satellite process. Threads in that case, make much more sense, as in case of failure they cause everything to terminate.

While thinking about it, we may not need threads at all though (like you said) but I'm not sure if mio is a good solution for this problem either. I will investigate this soon.